app/templates/main/serviceTranscript.html

```diff
@@ -4,102 +4,73 @@
 {% block app_content %}
 <div class="w-100 d-print-inline-block">
 <h1 class="text-center mt-5 mb-3">{{userData.firstName}} {{userData.lastName}}</h1>
-<button type="button" class="btn btn-primary float-end d-print-none" onclick="window.print()"><span class="bi bi-printer d-print-none"></span> Print</button>
-{% if programs.exists() %}
-  <h2 class="m-1">Programs</h2>
-  <table class="table table-bordered table-striped">
-    <tr>
-      <th>Program Name</th>
-      <th>Term</th>
-      <th>Accrued Hours</th>
-    </tr>
-    {% for program in programs %}
-    <tr>
-      <td> {{program.program.programName}}</td>
-      <td> {{program.event.term.description}}</td>
-      <td class="w-25"> {{program.hoursEarned}}</td>
-    </tr>
+{% if allEventTranscript %}
+    {% for program in allEventTranscript %}
+      {% set hours = allEventTranscript[program][0][1] %}
+      {% set terms = allEventTranscript[program] %}
+        {% if not (hours == None and terms|length == 1) %}
+          <div class="card-body">
+            {% if program == "CELTS Sponsored Events" %}
+                <h5 class="card-title">{{program}}</h5>
+            {% else %}
+                <h5 class="card-title">{{program.programName}}</h5>
+            {% endif %}
+            {% for item in terms %}
+                {% if item[1] != None%}
+                    <h6 class="card-subtitle mb-2 text-muted"> {{item[0]}} - {{item[1]|round(2)}} hour(s)</h6>
+                {% endif %}
+            {% endfor %}
+            {% if program != "CELTS Sponsored Events" %}
+                <p class="card-text">{{program.description}}</p>
+                <a class="d-print-none" href="{{program.url}}" class="card-link">More Information</a>
+            {% endif %}
+          </div>
+        {% endif %}
```

Comment:
It feels a little messy in this block. I think it's because we assume a lot about what the data format might be, and what indicates an empty list or not, or the program vs non program card.

That hour check is odd to me. Is there a way to avoid the triple array index and that particular boolean expression in the conditional? Like, why would there be a single term in the array if there were no hours for it?